test/unit/format/sam_file_format_test_template.hpp

```diff
@@ -0,0 +1,688 @@
+                                        std::remove_cvref_t<decltype(bio::map_io::field_types_sam<>)>>;
+    bio::map_io::reader_options default_options{};
+    bio::format_input_handler<bio::sam> input_handler{istream, default_options};
+    record_t rec;
+
+    for (unsigned i = 0; i < 3; ++i)
+    {
+        input_handler.parse_next_record_into(rec);
+
+        EXPECT_EQ(rec.id(),          this->ids[i])                << "failed at i = " << i << std::endl;
+        EXPECT_EQ(rec.flag(),        this->flags[i])              << "failed at i = " << i << std::endl;
+        EXPECT_EQ(rec.rname(),       this->ref_id)                << "failed at i = " << i << std::endl;
+        EXPECT_EQ(rec.pos(),         this->positions[i])          << "failed at i = " << i << std::endl;
+        EXPECT_EQ(rec.mapq(),        this->mapqs[i])              << "failed at i = " << i << std::endl;
+        EXPECT_RANGE_EQ(rec.cigar(), this->cigars[i]);
+        EXPECT_EQ(rec.rnext(),       this->ref_id)                << "failed at i = " << i << std::endl;
+        EXPECT_EQ(rec.pnext(),       std::get<1>(this->mates[i])) << "failed at i = " << i << std::endl;
+        EXPECT_EQ(rec.tlen(),        std::get<2>(this->mates[i])) << "failed at i = " << i << std::endl;
+        EXPECT_RANGE_EQ(rec.seq(),   this->seqs[i]);
+        EXPECT_RANGE_EQ(rec.qual(),  this->quals[i]);
+        // EXPECT_EQ(rec.tags(), this->tag_dicts[i]) << "failed at i = " << i << std::endl;
+    }
+}
+
+TYPED_TEST_P(sam_file_read, all_missing_data)
+{
+    typename TestFixture::stream_type istream{this->empty_input};
+
+    using record_t =
+      bio::detail::record_from_typelist<std::remove_cvref_t<decltype(bio::map_io::default_field_ids)>,
+                                        std::remove_cvref_t<decltype(bio::map_io::field_types_sam<>)>>;
+    bio::map_io::reader_options default_options{};
+    bio::format_input_handler<bio::sam> input_handler{istream, default_options};
+    record_t rec;
+
+    input_handler.parse_next_record_into(rec);
+
+    EXPECT_TRUE(rec.id().empty());
+    EXPECT_TRUE(rec.rname().empty());
+    EXPECT_TRUE(rec.rnext().empty());
+    EXPECT_TRUE(rec.cigar().empty());
+    EXPECT_TRUE(rec.seq().empty());
+    EXPECT_TRUE(rec.qual().empty());
+    // EXPECT_TRUE(rec.tags().empty()) << tags;
+
+    EXPECT_EQ(rec.flag(), bio::map_io::sam_flag{0u});
+    EXPECT_EQ(rec.pos(), 0);
+    EXPECT_EQ(rec.pnext(), 0);
+    EXPECT_EQ(rec.mapq(), 0u);
+    EXPECT_EQ(rec.tlen(), 0);
+}
+
+TYPED_TEST_P(sam_file_read, select_fields)
+{
+    typename TestFixture::stream_type istream{this->empty_input};
+
+    constexpr auto fid = bio::vtag<bio::field::rname, bio::field::pos>;
+    constexpr auto ftype = bio::ttag<std::string_view, int64_t>;
+
+    using record_t =
+      bio::detail::record_from_typelist<std::remove_cvref_t<decltype(fid)>, std::remove_cvref_t<decltype(ftype)>>;
+    bio::map_io::reader_options default_options{};
```

Comment:
```suggestion
    bio::map_io::reader_options default_options{.field_ids = fid, .field_types = ftype};
```

You didn't actually select the fields, so it cannot work